homeassistant/components/denonavr/config_flow.py

```diff
@@ -0,0 +1,220 @@
+
+        if not self.serial_number:
+            _LOGGER.error(
+                "Could not get serial number of host %s, "
+                "using the mac_address as identification",
+                self.host,
+            )
+            if not mac_address:
+                return self.async_abort(reason="no_mac")
+            self.serial_number = mac_address
+
+        unique_id = self.construct_unique_id(self.model_name, self.serial_number)
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=receiver.name,
+            data={
+                CONF_HOST: self.host,
+                CONF_MAC: mac_address,
+                CONF_TIMEOUT: self.timeout,
+                CONF_SHOW_ALL_SOURCES: self.show_all_sources,
+                CONF_ZONE2: self.zone2,
+                CONF_ZONE3: self.zone3,
+                CONF_TYPE: receiver.receiver_type,
+                CONF_MODEL: self.model_name,
+                CONF_MANUFACTURER: receiver.manufacturer,
+            },
+        )
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Denon AVR.
+
+        This flow is triggered by the SSDP component. It will check if the
+        host is already configured and delegate to the import step if not.
+        """
+        # Filter out non-Denon AVRs#1
+        if (
+            discovery_info.get(ssdp.ATTR_UPNP_MANUFACTURER)
+            not in SUPPORTED_MANUFACTURERS
+        ):
+            return self.async_abort(reason="not_denonavr_manufacturer")
+
+        # Check if required information is present to set the unique_id
+        if (
+            ssdp.ATTR_UPNP_MODEL_NAME not in discovery_info
+            or ssdp.ATTR_UPNP_SERIAL not in discovery_info
+        ):
+            return self.async_abort(reason="not_denonavr_missing")
+
+        self.model_name = discovery_info[ssdp.ATTR_UPNP_MODEL_NAME].replace("*", "")
+        self.serial_number = discovery_info[ssdp.ATTR_UPNP_SERIAL]
+        self.host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+
+        unique_id = self.construct_unique_id(self.model_name, self.serial_number)
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured({CONF_HOST: self.host})
+
+        return await self.async_step_settings()
+
+    @staticmethod
+    def construct_unique_id(model_name, serial_number):
+        """Construct the unique id from the ssdp discovery or user_step."""
+        return f"{model_name}-{serial_number}"
+
+    async def async_get_mac(self, host):
+        """Get the mac address of the DenonAVR receiver."""
+        mac_address = await self.hass.async_add_executor_job(
+            partial(get_mac_address, **{"ip": host})
+        )
+        if not mac_address:
+            mac_address = await self.hass.async_add_executor_job(
+                partial(get_mac_address, **{"hostname": host})
+            )
```

Comment:
`get_mac_address` can throw an exception here

Reply:
What schould I do to resolve that?
Do you know what kind of exception, than I can put a try, except around it?

Reply:
It cat at least throw `OSError` and probably a lot of others.  You'll probably need a broad check for `Exception` here.

Reply:
alright I put a broad try except around it and made sure the mac_addres = None if we get an exception (and I log the exception), in that way we return self.async_abort(reason="no_mac").

Is this okay?

Reply:
According to the docstring it will except exceptions silently and return `None`:
https://github.com/GhostofGoes/getmac/blob/3569b7ce503aa94f81a2d4cbab4a3aea8f44e39e/getmac/getmac.py#L119

Reply:
So schould I then remove the try except again?

Reply:
> So schould I then remove the try except again?

I've definitely that this throw on me before unexpectedly.  Let me take a look at the history to see if they fixed the issue recently as Martin's comment points out that they expect they have caught all exceptions possible.

Reply:
It doesn't look like they have fixed the issue upstream.  I've found at least 3 places an exception could be thrown unexpectedly.

Reply:
so put back the try catch?

Reply:
I recall the issue was with `nmap_tracker`, but I had not realized at the time `getmac` was expected to trap exceptions as I would have saved the backtrace and already reported the issue upstream.

It needs to be fixed at some point since it is a bug in upstream code given the docstring.  Its likely going to be a rare case unless they have ipv6 disabled (then it will always happen) so there are two options here:

1. Put back in the try catch to insulate us from the unexpected behavior. The downside is this sweeps the problem under the rug but favors stability.

2. Let to blow up and report the issue upstream when/if we get an issue about it.  The downside here is the issue probably won't get fixed unless someone else reported it.  

It would be nice to get it fixed upstream, but I expect we need a backtrace for that to happen.  If you are going to be able to respond quickly to a report of an unexpected exception, report upstream, and get a fix out, go with option 2.  Otherwise go with option 1.

Reply:
I no almost nothing about the whole getmac lib, so I will just use the try catch, better save than sorry :)